thriftbp/server.go

```diff
@@ -69,20 +69,22 @@ type ServerConfig struct {
 // and protocol to serve the given TProcessor which is wrapped with the
 // given ProcessorMiddlewares.
 func NewServer(cfg ServerConfig) (*thrift.TSimpleServer, error) {
-	var transport *thrift.TServerSocket
+
+	// Leaving this default socket initialization intact so that we
+	// don't break existing usages of NewServer. Ideally, by the time
+	// we've reached NewServer, all necessary configuration should
+	// be completed. If we are ever able to deprecate this use case
+	// we will be able to fully remove cfg.Addr and serverOptAddress
+	// as well.
 	if cfg.Socket == nil {
```

Comment:
In sourcegraph I was only able to find two place sites where removing this would break user code.